lapack/testlapack/dlangb.go

```diff
@@ -0,0 +1,110 @@
+// Copyright ©2021 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package testlapack
+
+import (
+	"fmt"
+	"math"
+	"testing"
+
+	"golang.org/x/exp/rand"
+
+	"gonum.org/v1/gonum/floats"
+	"gonum.org/v1/gonum/lapack"
+)
+
+type Dlangber interface {
+	Dlangb(norm lapack.MatrixNorm, m, n, kl, ku int, ab []float64, ldab int, work []float64) float64
+}
+
+func DlangbTest(t *testing.T, impl Dlangber) {
+	rnd := rand.New(rand.NewSource(1))
+	for _, norm := range []lapack.MatrixNorm{lapack.MaxAbs, lapack.MaxRowSum, lapack.MaxColumnSum, lapack.Frobenius} {
+		t.Run(normToString(norm), func(t *testing.T) {
+			for _, n := range []int{0, 1, 2, 3, 4, 5, 10} {
+				for _, m := range []int{0, 1, 2, 3, 4, 5, 10} {
+					for _, kl := range []int{0, 1, 2, 3, 4, 5, 10} {
+						for _, ku := range []int{0, 1, 2, 3, 4, 5, 10} {
+							for _, ldab := range []int{kl + ku + 1, kl + ku + 1 + 7} {
+								for iter := 0; iter < 10; iter++ {
+									dlangbTest(t, impl, rnd, norm, m, n, kl, ku, ldab)
+								}
+							}
+						}
+					}
+				}
+			}
+		})
+	}
+}
+
+func dlangbTest(t *testing.T, impl Dlangber, rnd *rand.Rand, norm lapack.MatrixNorm, m, n, kl, ku, ldab int) {
+	const tol = 1e-14
+
+	name := fmt.Sprintf("m=%v,n=%v,kl=%v,ku=%v,ldab=%v", m, n, kl, ku, ldab)
+
+	// Generate a random band matrix.
+	ab := randomSlice(m*ldab, rnd)
+	// Sometimes put a NaN into the matrix.
+	if m > 0 && n > 0 && rnd.Float64() < 0.5 {
+		i := rnd.Intn(m)
+		ab[i*ldab+kl] = math.NaN()
+	}
```

Comment:
internal/testrand would be useful here, but the `extreme` type is not exported. Is it worth exporting that so that it's not just configurable in the command line and is more generally usable? (separate PR)

Reply:
Looking into how internal/testrand could be used in testlapack is on my todo list ...